projects/wireshark/build.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash -eu
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Wireshark build.sh script inspired from projects/gnutls/build.sh by Alex Gaynor.
+
+FUZZ_DISSECTORS="ip \
+  udp"
+
+export WIRESHARK_LIBS_PATH="$SRC/libs"
+mkdir -p "$WIRESHARK_LIBS_PATH"
+
+# compile static version of libs
+# XXX, with static wireshark linking each fuzzer binary is ~238 MB (just libwireshark.a is 423 MBs).
+# XXX, wireshark build infrastucture is not ready for including plugins into static binary.
+CONFOPTS="--disable-shared --enable-static --without-plugins"
+
+# disable optional dependencies
+CONFOPTS="$CONFOPTS --without-pcap --without-ssl --without-gnutls"
+
+# need only libs, disable programs
+CONFOPTS="$CONFOPTS --disable-wireshark --disable-tshark --disable-sharkd \
+             --disable-dumpcap --disable-capinfos --disable-captype --disable-randpkt --disable-dftest \
+             --disable-editcap --disable-mergecap --disable-reordercap --disable-text2pcap \
+             --without-extcap \
+         "
+
+./autogen.sh
+./configure --prefix="$WIRESHARK_LIBS_PATH" $CONFOPTS --disable-warnings-as-errors
+
+make "-j$(nproc)"
+make install
+
+WIRESHARK_FUZZERS_COMMON_FLAGS="-lFuzzingEngine /usr/local/lib/libc++.a \
+    -"L$WIRESHARK_LIBS_PATH/lib" -lwireshark -lwiretap -lwsutil \
+    `pkg-config --libs glib-2.0` -lgcrypt -lgpg-error -lz"
+
+for dissector in $FUZZ_DISSECTORS; do
+  fuzzer_name=fuzzshark_dissector_${dissector}
+
+  # -I$SRC/wireshark is correct, wireshark don't install include files.
+  $CC $CFLAGS -I $SRC/wireshark/ `pkg-config --cflags glib-2.0` \
+      $SRC/wireshark/tools/oss-fuzzshark.c \
+      -c -o /tmp/${fuzzer_name}.o \
+      -DFUZZ_DISSECTOR_TARGET=\"$dissector\"
+
+  $CXX $CXXFLAGS /tmp/${fuzzer_name}.o \
+      -o $OUT/${fuzzer_name} \
+      ${WIRESHARK_FUZZERS_COMMON_FLAGS}
+
+  # XXX, wireshark leaks memory - disable leaks checking
+  echo -en "[libfuzzer]\nmax_len = 1024\ndetect_leaks=0\n" > $OUT/${fuzzer_name}.options
```

Comment:
Problem is fixed, but right now fuzzer (oss-fuzzshark) is still stateful - like rest of wireshark.
We can (and I think we should) disable at least IP, TCP (and other) reassembly for fuzzers, but it's not done yet, I hope it's not a blocker?